feat: Normal user list (#7887)
owncloud/ocis: changelog/unreleased/enhancement-allow-listing-regular-users.md

Enhancement: Allow regular users to list other users

> Regular users can search for other users, but having some limitations

Reviewer: ```suggestion
Regular users can search for other users, with some limitations.
```

Reviewer: Maybe list the limitations in the changelog item as it has high impact

Reviewer 2: done
